lua/pac3/editor/server/wear.lua

```diff
@@ -108,68 +122,66 @@ duplicator.RegisterEntityModifier("pac_config", function(ply, ent, parts)
 	end)
 end)
 
-function pace.SubmitPart(data, filter)
-	if istable(data.part) then
-		if last_frame == frame_number then
-			table.insert(pace.StreamQueue, {data, filter})
-			pace.dprint("queuing part %q from %s", data.part.self.Name, tostring(data.owner))
-			return "queue"
-		end
+function pace.SubmitPartNow(data, filter)
+	local part = data.part
+	local owner = data.owner
+
+	-- last arg "true" is pac3 only in case you need to do your checking differently from pac2
+	local allowed, reason = hook.Run("PrePACConfigApply", owner, data, true)
+	if allowed == false then return allowed, reason end
```

Comment:
I moved this `allowed == false` check up to where we run the hook. I don't understand why we ran the next block of code first, it doesn't seem necessary.

Please correct me if I'm wrong on that.